app/api/relationships.v2/database/specs/MongoRelationshipsDataSource.spec.ts

```diff
@@ -259,3 +260,25 @@ describe('getAll()', () => {
     expect(rels).toHaveLength(9);
   });
 });
+
+describe('getByefinition()', () => {
+  it('should find the relationships from sourceEntity to targetEntity of the provided type', async () => {
+    const rels = await ds
+      .getByDefinition('entity1', factory.id('nullType').toString(), 'hub1')
+      .all();
+    expect(rels).toEqual([expect.objectContaining({ _id: factory.id('rel1').toString() })]);
+    rels.forEach(rel => {
+      expect(rel).toBeInstanceOf(Relationship);
+    });
+  });
+
+  it('should only find relationships that are not text references', async () => {
```

Comment:
That's exactly why. I anyways think this needs conceptual work in the future, but it's the safest path. If you cannot create text references through this field, you should not be able to delete them. They will anyway show as results of the query and be denormalized.
This could also be better treated in a validation and the UI, but I would wait for that until we start working on the frontend and know what we want to achieve.